src/formatDate.js

```diff
@@ -1,5 +1,39 @@
 'use strict';
 
+/**
+ * @param {Object} datePartsMatched
+ * @param {string[]} toFormat
+ * @param {string[]} fromFormat
+ *
+ * @returns {Object}
+ */
+
+function convertYear(datePartsMatched, fromFormat, toFormat) {
+  const FOUR_DIGIT_YEAR_FORMAT = 'YYYY';
+  const TWO_DIGIT_YEAR_FORMAT = 'YY';
+
+  const datePartsMatchedCopy = { ...datePartsMatched };
+
+  if (
+    fromFormat.includes(FOUR_DIGIT_YEAR_FORMAT) &&
+    toFormat.includes(TWO_DIGIT_YEAR_FORMAT)
+  ) {
+    const year = datePartsMatchedCopy[FOUR_DIGIT_YEAR_FORMAT];
+
+    datePartsMatchedCopy[TWO_DIGIT_YEAR_FORMAT] = year.slice(2);
+  } else if (
+    fromFormat.includes(TWO_DIGIT_YEAR_FORMAT) &&
+    toFormat.includes(FOUR_DIGIT_YEAR_FORMAT)
+  ) {
+    const year = datePartsMatchedCopy[TWO_DIGIT_YEAR_FORMAT];
+    const newYear = +year < 30 ? `20${year}` : `19${year}`;
+
+    datePartsMatchedCopy[FOUR_DIGIT_YEAR_FORMAT] = newYear;
+  }
+
+  return datePartsMatchedCopy;
```

Comment:
The `convertYear` function is a bit misleading in terms of variable naming. The `datePartsMatched` object does not actually contain 'matched' parts but rather a mapping from format strings to their respective values. Consider renaming it to `datePartsMap` or similar to better reflect its purpose.